### PR TITLE
[SPARK-42711][BUILD]Update usage info and shellcheck warn/error fix for build/sbt tool

### PR DESCRIPTION
**Pull request by liang3zy22**
### What changes were proposed in this pull request?

The build/sbt script's usage information have several error. See the info below:
```
(base) spark% ./build/sbt -help
Usage:  [options]

  -h | -help         print this message
  -v | -verbose      this runner is chattier

```
There is no script name after the usage. With this change, the info become:
```
(base) spark % ./build/sbt -help
Usage: sbt [options]

  -h | -help         print this message
  -v | -verbose      this runner is chattier

```
This changes also fix several shellcheck error.
 
### Why are the changes needed?

Bug fix.

### Does this PR introduce _any_ user-facing change?

Yes, the user of build/sbt tool will see usage information updated.

### How was this patch tested?

Manual check the script codes.

### CODE REVIEW

**Comment by yaooqinn** [comment]
This script is just a fork from the sbt repo, and the hint info for current java seems correct to me. FYI, https://github.com/sbt/sbt/blob/1.9.x/sbt#L576

**Comment by liang3zy22** [comment]
Thanks for comment, @yaooqinn .  The build/sbt script seems to be a simplified sbt script. I think the hint info for current java is not correct.to be honest, I think it is a copy-paste error from the hint info for sbt_version argument. I can create a PR in sbt/sbt project to see if it can be updated.

File: build/sbt [hunk 1]
```diff
@@ -17,7 +17,7 @@
 # limitations under the License.
 #
 
-SELF=$(cd $(dirname $0) && pwd)
+SELF=$(cd "$(dirname "$0")" && pwd)
```
Comment (srowen) [inline build/sbt:20]: Are these changes necessary/related?

File: build/sbt [hunk 2]
```diff
@@ -62,7 +63,7 @@ Usage: $script_name [options]
   -sbt-rc                   use an RC version of sbt
   -sbt-snapshot             use a snapshot version of sbt
 
-  # java version (default: java from PATH, currently $(java -version 2>&1 | grep version))
+  # JAVA_HOME (this will override default JAVA_HOME)
```
Comment (srowen) [inline build/sbt:66]: I'm not sure I understand your new comment. What is this telling users? it overrides JAVA_HOME, but it says JAVA_HOME overrides JAVA_HOME .. ?

**Review by srowen — CHANGES_REQUESTED** [review]

File: build/sbt (hunk 1 above)
Comment (liang3zy22) [reply build/sbt:20]: This is shellcheck error/warning fix. I think we can keep it.

File: build/sbt (hunk 2 above)
Comment (liang3zy22) [reply build/sbt:66]: Well, my words are not clear. I will rollback to old hint info.

**Comment by srowen** [comment]
OK, so this is just changing some stuff for shellcheck. It seems OK. This is just a copy of the sbt script. Normally I'd say, does this make it harder to reason about changes if we pull a new copy from upstream? but probably not. This also probably doesn't fix any actual problem. I'm neutral on bothering to change these.

**Comment by liang3zy22** [comment]
Yes, this PR is kind of useless. I close it now.
